canonical-web-and-design/maas-ui
# feat(ui): rename LxdProject to LXDHostVMs and make reusable

## Done

- Renamed `LxdProject` to `LXDHostVMs`. It's currently only used for `LXDSingleDetails` but will also be used in `LXDClusterDetails` when a LXD host is selected.
- Renamed `ProjectSummaryCard` to `LXDVMsSummaryCard`, included a VF section and made it presentational. It also now shows the little cards by default, though they're still using the old design. It's currently only used in `LXDHostVMs` but will also be able to be used in the component that renders cluster VMs
- Renamed `LxdVMs` to `LXDVMsTable` which is a bit clearer now that there are a lot of "VM" components

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Check that the "Virtual machines" tab of a KVM now includes a "VF" section in the summary card
- Check that the "Resources" cards are unchanged

## Fixes

Fixes canonical-web-and-design/app-squad#360

## Screenshot
### LXDVMsSummaryCard
![Screenshot 2021-10-07 at 12-43-20 KVM bolla MAAS](https://user-images.githubusercontent.com/25733845/136311914-88bec3f7-37d4-40bf-9c8a-c9deb197562a.png)
